shared/src/commonMain/kotlin/com/example/tasky/dataSource/AgendaDataSource.kt

```diff
@@ -57,10 +64,27 @@ class AgendaDataSource(
             }
         }
 
-    suspend fun updateEvent(body: UpdateEventBody): ResultWrapper<Unit, BaseError> =
+    suspend fun updateEvent(body: UpdateEventBody): ResultWrapper<Event, BaseError> =
         safeCall {
             httpClient.put(EventPath()) {
-                setBody(body)
+                setBody(
+                    MultiPartFormDataContent(
+                        formData {
+                            append("id", body.id)
+                            append("title", body.title)
+                            append("description", body.description)
+                            append("from", body.from)
+                            append("to", body.to)
+                            append("remindAt", body.remindAt)
+                            append("attendeeIds", body.attendeeIds)
+                            append("deletedPhotoKeys", body.deletedPhotoKeys)
+                            append("isGoing", body.isGoing)
+                            body.photos.forEachIndexed { index, bytes ->
+                                append("photo$index", bytes)
+                            }
+                        },
+                    ),
+                )
```

Comment:
A simpler approach would be using `submitFormWithBinaryData`, this is how you can set it up:

```
// create json from request domain model
val updateEventJson = Json.encodeToString(
  event.toUpdateEventRequest(
    isGoing = isGoing,
    deletedPhotoKeys = deletedPhotoKeys,
  )
)

// start of form request
httpClient.submitFormWithBinaryData(
  url = constructRoute("/event”), 

  // start of form
  formData = formData {

    // adding images
    photos.forEachIndexed { index, photoByteArray ->
      append("photo$index”, photoByteArray, Headers.build {
          append(HttpHeaders.ContentType, "image/jpeg")
          append(HttpHeaders.ContentDisposition, "filename=${event.id}_picture$index.jpg")
        }
      )
    }

    // adding body json
    append("update_event_request”, updateEventJson, Headers.build {
        append(HttpHeaders.ContentType, "text/plain")
        append(HttpHeaders.ContentDisposition, "form-data; name=\"update_event_request\"")
      }
    )
  }
) {
  // set the HTTP method to Put
  method = HttpMethod.Put
}
```